### PR TITLE
Leaner ZIOSpecWithKafka

### PR DESCRIPTION
**Pull request by erikvanoosten**
The new ZIOSpecWithKafka (from #528) is very useful for testing zio-kafka applications, however, it is extended with additional tools for naming which are not useful for everyone. Even so, it does force you to implement `kafkaPrefix`. This PR moves the naming extensions back to the zio-kafka-test module.

### CODE REVIEW

**Comment by erikvanoosten** [comment]
@tewecske You may want to review this PR since you proposed moving ZIOSpecWithKafka to zio-kafka-test-utils in #528.

**Comment by tewecske** [comment]
> @tewecske You may want to review this PR since you proposed moving ZIOSpecWithKafka to zio-kafka-test-utils in #528.

Sorry, I can't review it.
I agree that the abstract `kafkaPrefix` is not good like this in the utils but the random generators are somewhat useful I think.

File: zio-kafka-test/src/test/scala/zio/kafka/ZIOKafkaSpec.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package zio.kafka
+
+import zio.{ Task, ZIO }
+
+import java.util.UUID
+
+trait ZIOKafkaSpec extends ZIOSpecWithKafka {
```
[the remark this answers is not in the document]
Comment (erikvanoosten) [reply zio-kafka-test/src/test/scala/zio/kafka/ZIOKafkaSpec.scala:7]: Do you think it is generic enough? IMHO the naming generating utils are rather trivial and not really Kafka specific. Also, (again IMHO) it is bad practice to use random values in a test as it makes re-producing a failure harder (unless you are doing something like scalaspec with error minimization).

Having said all that, yes, that is an option and you want me to change it I will.

[the remark this answers is not in the document]
Comment (svroonland) [reply zio-kafka-test/src/test/scala/zio/kafka/ZIOKafkaSpec.scala:7]: I agree with @erikvanoosten that these are a bit too specific to include in a zio-kafka testing library for end users.

**Review by guizmaii — APPROVED** [review]

File: zio-kafka-test/src/test/scala/zio/kafka/ZIOKafkaSpec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (guizmaii) [reply zio-kafka-test/src/test/scala/zio/kafka/ZIOKafkaSpec.scala:7]: Ok I'll follow your opinions then mates :)

Anyway, if someone needs it, he/she can make a PR to move it or just copy this trait in his/her codebase

[the remark this answers is not in the document]
Comment (erikvanoosten) [reply zio-kafka-test/src/test/scala/zio/kafka/ZIOKafkaSpec.scala:7]: 🚀

**Pull request merged**